core/src/main/java/org/apache/calcite/plan/RelOptMaterializations.java

```diff
@@ -64,6 +73,9 @@
         getApplicableMaterializations(rel, materializations);
     final List<Pair<RelNode, List<RelOptMaterialization>>> applied =
         new ArrayList<>();
+    if (SubstitutionVisitor.disabled) {
```

Comment:
This is added for testing and differentiate MV optimization by `SubstitutionVisitor` and `AbstractMaterializedViewRule`. I'm not sure if I should add a config of `CalciteConnectionProperty`.
Please share comments if there's suggestion